web-bluetooth/scan.js

```diff
@@ -30,6 +30,16 @@ async function onButtonClick() {
 
     navigator.bluetooth.addEventListener('advertisementreceived', event => {
       log('Advertisement received.');
+      log('  Device ID: ' + event.device.id);
+      log('  RSSI: ' + event.rssi);
+      log('  TX Power: ' + event.txPower);
+      log('  UUIDs: ' + event.uuids);
+      event.manufacturerData.forEach((_, key) => {
+        log('  Manufacturer Data: ' + key);
+      });
+      event.serviceData.forEach((_, key) => {
+        log('  Service Data: ' + key);
+      });
```

Comment:
If dumping the keys, we might as well read out the value bytes as well.  (They'll never get too long, as ad packets even in Bluetooth 5 is 255 bytes, but most devices will just be less than a dozen.)

Something like:

...Data.forEach((valueDataView, key) => {
log.' ...Data: ' + key + ': ' + (new Uint8Array(valueDataView.buffer).toString()

That should get things out in decimal.. a tad nicer might be to print in hex https://stackoverflow.com/questions/40031688/javascript-arraybuffer-to-hex

Super fancy would be an option drop down in the sample that chose to print as decimal, hex, or ASCII characters.